app/src/main/java/dev/arkbuilders/rate/data/db/PairAlertRepoImpl.kt

```diff
@@ -0,0 +1,92 @@
+package dev.arkbuilders.rate.data.db
+
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import dev.arkbuilders.rate.domain.model.CurrencyCode
+import dev.arkbuilders.rate.domain.model.PairAlert
+import dev.arkbuilders.rate.domain.repo.PairAlertRepo
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import java.time.OffsetDateTime
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Entity
+data class RoomPairAlert(
+    @PrimaryKey(autoGenerate = true)
+    val id: Long,
+    val targetCode: CurrencyCode,
+    val baseCode: CurrencyCode,
+    val targetPrice: Double,
+    val startPrice: Double,
+    val alertPercent: Double?,
+    val oneTimeNotRecurrent: Boolean,
+    val enabled: Boolean,
+    val lastDateTriggered: String?,
+    val group: String?
+)
+
+@Dao
+interface PairAlertDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(pairAlert: RoomPairAlert): Long
+
+    @Query("SELECT * FROM RoomPairAlert where id = :id")
+    suspend fun getById(id: Long): RoomPairAlert?
+
+    @Query("SELECT * FROM RoomPairAlert")
+    suspend fun getAll(): List<RoomPairAlert>
+
+    @Query("SELECT * FROM RoomPairAlert")
+    fun getAllFlow(): Flow<List<RoomPairAlert>>
```

Comment:
Could you please elaborate this? When do we need a pure list and when a flow?

Reply:
Flow is needed on screen. Just list is needed when checking pair alerts in background (`CurrencyMonitorWorker`).